art/estimators/speech_recognition/tensorflow_lingvo.py

```diff
@@ -0,0 +1,516 @@
+            "basename": "wpm_16k_librispeech.vocab",
+        },
+        "decoder": {
+            "uri": (
+                "https://raw.githubusercontent.com/hesseltuinhof/lingvo/qin_patched_decoder/lingvo/tasks/asr/decoder.py"
+            ),
+            "basename": "decoder_patched.py",
+        },
+    }
+
+    def __init__(
+        self,
+        clip_values: Optional["CLIP_VALUES_TYPE"] = None,
+        channels_first: Optional[bool] = None,
+        preprocessing_defences: Union["Preprocessor", List["Preprocessor"], None] = None,
+        postprocessing_defences: Union["Postprocessor", List["Postprocessor"], None] = None,
+        preprocessing: "PREPROCESSING_TYPE" = None,
+        random_seed: Optional[int] = None,
+        sess: Optional["Session"] = None,
+    ):
+        """
+        Initialization.
+
+        :param clip_values: Tuple of the form `(min, max)` of floats or `np.ndarray` representing the minimum and
+               maximum values allowed for features. If floats are provided, these will be used as the range of all
+               features. If arrays are provided, each value will be considered the bound for a feature, thus
+               the shape of clip values needs to match the total number of features.
+        :param channels_first: Set channels first or last.
+        :param preprocessing_defences: Preprocessing defence(s) to be applied by the classifier.
+        :param postprocessing_defences: Postprocessing defence(s) to be applied by the classifier.
+        :param preprocessing: Tuple of the form `(subtrahend, divisor)` of floats or `np.ndarray` of values to be
+            used for data preprocessing. The first value will be subtracted from the input. The input will then
+               be divided by the second one.
+        :param random_seed: Specify a random seed.
+        """
+        import pkg_resources
+
+        import tensorflow.compat.v1 as tf1
+
+        # Super initialization
+        super().__init__(
+            clip_values=clip_values,
+            channels_first=channels_first,
+            preprocessing_defences=preprocessing_defences,
+            postprocessing_defences=postprocessing_defences,
+            preprocessing=preprocessing,
+        )
+        self.random_seed = random_seed
+        if self.preprocessing is not None:
+            raise ValueError("This estimator does not support `preprocessing`.")
+        if self.preprocessing_defences is not None:
+            raise ValueError("This estimator does not support `preprocessing_defences`.")
+        if self.postprocessing_defences is not None:
+            raise ValueError("This estimator does not support `postprocessing_defences`.")
+
+        # check required TensorFlow version
+        assert tf1.__version__ == "2.1.0", "The Lingvo estimator only supports TensorFlow 2.1.0."
+
+        # check required Python version
+        assert sys.version_info[:2] == (3, 6), "The Lingvo estimator only supports Python 3.6."
```

Comment:
Should we transform these checks into `if` and `raise` exceptions to avoid users running into unexpected behavior with `-O` flag?